File: DiKErnel/Test/DiKErnel.DomainLibrary.Test/Defaults/GrassRevetmentWaveRunup/GrassRevetmentWaveRunupDefaultsTest.cpp
// Copyright (C) Stichting Deltares 2022. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include <gtest/gtest.h>

#include "GrassRevetmentWaveRunupDefaults.h"

namespace DiKErnel::DomainLibrary::Test
{
    TEST(GrassRevetmentWaveRunupDefaultsTest, GetIncreasedLoadTransitionAlphaM_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1, GrassRevetmentWaveRunupDefaults::GetIncreasedLoadTransitionAlphaM());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetReducedStrengthTransitionAlphaS_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1, GrassRevetmentWaveRunupDefaults::GetReducedStrengthTransitionAlphaS());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetAverageNumberOfWavesCtm_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(0.92, GrassRevetmentWaveRunupDefaults::GetAverageNumberOfWavesCtm());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetRepresentativeWaveRunup2PAru_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1.65, GrassRevetmentWaveRunupDefaults::GetRepresentativeWaveRunup2PAru());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetRepresentativeWaveRunup2PBru_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(4, GrassRevetmentWaveRunupDefaults::GetRepresentativeWaveRunup2PBru());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetRepresentativeWaveRunup2PCru_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1.5, GrassRevetmentWaveRunupDefaults::GetRepresentativeWaveRunup2PCru());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetRepresentativeWaveRunup2PGammab_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1, GrassRevetmentWaveRunupDefaults::GetRepresentativeWaveRunup2PGammab());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetRepresentativeWaveRunup2PGammaf_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(1, GrassRevetmentWaveRunupDefaults::GetRepresentativeWaveRunup2PGammaf());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetWaveAngleImpactAbeta_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(0.0022, GrassRevetmentWaveRunupDefaults::GetWaveAngleImpactAbeta());
    }

    TEST(GrassRevetmentWaveRunupDefaultsTest, GetWaveAngleImpactBetamax_ExpectedValue)
    {
        // Call& Assert
        ASSERT_DOUBLE_EQ(80, GrassRevetmentWaveRunupDefaults::GetWaveAngleImpactBetamax());
    }
}

File: Test/DiKErnel.DomainLibrary.Test/Defaults/GrassRevetmentWaveRunup/GrassRevetmentWaveRunupDefaultsTest.cs
// Copyright (C) Stichting Deltares and State of the Netherlands 2023. All rights reserved.
//
// This file is part of DiKErnel.
//
// This program is free software; you can redistribute it and/or modify it under the terms of
// the GNU Lesser General Public License as published by the Free Software Foundation; either
// version 3 of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program; if not, see <https://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

using DiKErnel.DomainLibrary.Defaults.GrassRevetmentWaveRunup;
using NUnit.Framework;

namespace DiKErnel.DomainLibrary.Test.Defaults.GrassRevetmentWaveRunup
{
    [TestFixture]
    public class GrassRevetmentWaveRunupDefaultsTest
    {
        [Test]
        public void RepresentativeWaveRunup2PAru_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.RepresentativeWaveRunup2PAru, Is.EqualTo(1.65));
        }

        [Test]
        public void RepresentativeWaveRunup2PBru_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.RepresentativeWaveRunup2PBru, Is.EqualTo(4));
        }

        [Test]
        public void RepresentativeWaveRunup2PCru_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.RepresentativeWaveRunup2PCru, Is.EqualTo(1.5));
        }

        [Test]
        public void RepresentativeWaveRunup2PGammab_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.RepresentativeWaveRunup2PGammab, Is.EqualTo(1));
        }

        [Test]
        public void RepresentativeWaveRunup2PGammaf_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.RepresentativeWaveRunup2PGammaf, Is.EqualTo(1));
        }

        [Test]
        public void WaveAngleImpactAbeta_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.WaveAngleImpactAbeta, Is.EqualTo(0.0022));
        }

        [Test]
        public void WaveAngleImpactBetamax_ExpectedValue()
        {
            // Call & Assert
            Assert.That(GrassRevetmentWaveRunupDefaults.WaveAngleImpactBetamax, Is.EqualTo(80));
        }
    }
}